src/views/home/components/amount-input/amount-input.view.tsx

```diff
@@ -0,0 +1,85 @@
+import { BigNumber } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import { ChangeEvent, FC, useState } from "react";
+
+import useAmountInputtStyles from "src/views/home/components/amount-input/amount-input.styles";
+import Typography from "src/views/shared/typography/typography.view";
+import { Token } from "src/domain";
+
+interface onChangeParams {
+  amount: BigNumber;
+  isInvalid: boolean;
+}
+
+interface AmountInputProps {
+  token: Token;
+  balance: BigNumber;
+  fee: BigNumber;
+  onChange: (params: onChangeParams) => void;
+}
+
+const AmountInput: FC<AmountInputProps> = ({ token, balance, fee, onChange }) => {
+  const [value, setValue] = useState("");
+  const classes = useAmountInputtStyles(value.length);
+  const actualFee = token.symbol.includes("ETH") ? fee : BigNumber.from(0);
+
+  const getFixedTokenAmount = (amount: string, decimals = 18): string => {
```

Comment:
Why this default? I'd make the param required so the consumer is aware of the feature.